apps/src/accounts/ChangeEmail/ChangeEmailForm.jsx

```diff
@@ -0,0 +1,133 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import FontAwesomeV6Icon from '@cdo/apps/componentLibrary/fontAwesomeV6Icon/FontAwesomeV6Icon';
+import Link from '@cdo/apps/componentLibrary/link/Link';
+import {RadioButtonsGroup} from '@cdo/apps/componentLibrary/radioButton';
+import TextField from '@cdo/apps/componentLibrary/textField/TextField';
+import Typography from '@cdo/apps/componentLibrary/typography/Typography';
+import i18n from '@cdo/locale';
+
+import {pegasus} from '../../lib/util/urlHelpers';
+
+import styles from './formStyle.module.scss';
+
+export default class ChangeEmailForm extends React.Component {
+  static propTypes = {
+    values: PropTypes.shape({
+      newEmail: PropTypes.string,
+      currentPassword: PropTypes.string,
+      emailOptIn: PropTypes.string,
+    }).isRequired,
+    validationErrors: PropTypes.shape({
+      newEmail: PropTypes.string,
+      currentPassword: PropTypes.string,
+      emailOptIn: PropTypes.string,
+    }).isRequired,
+    userType: PropTypes.oneOf(['teacher', 'student']).isRequired,
```

Comment:
Nit: For uses of 'teacher' and 'student', would you mind using the `UserTypes` constant from `@cdo/generated-scripts/sharedConstants` (doesn't really make a big difference so totally fine either way)

Reply:
ooo, for sure! thanks for catching this!

Reply:
this turned into some crazy cryptic typescript, but it works! 🙌 

https://github.com/code-dot-org/code-dot-org/pull/61541/files#diff-9199bdcec136a24b1e343656cb93c5da371e0111ef68dd204a573cbec7bd5c0bR1-R3